### PR TITLE
ADEN-8517 Disable outstream bidding if featured video is on the page

### PR DESCRIPTION
**Pull request by dmnsobczak**

### CODE REVIEW

File: extensions/wikia/AdEngine3/src/setup.js [hunk 1]
```diff
@@ -179,6 +179,13 @@ function setupAdContext(wikiContext, isOptedIn = false, geoRequiresConsent = tru
 		if (!isGeoEnabled('wgAdDriverPubMaticOutstreamCountries')) {
 			context.remove('bidders.prebid.pubmatic.slots.INCONTENT_PLAYER');
 		}
+
+		const unbiddedSlotName = hasFeaturedVideo ? 'INCONTENT_PLAYER' : 'FEATURED';
+		let videoBidders = ['appnexusAst', 'beachfront', 'lkqd', 'pubmatic', 'rubicon'];
```
Comment (jacekbj) [inline extensions/wikia/AdEngine3/src/setup.js]: Can we mark bidders as "video" bidders in `ad-context.js` and draw the list from there? Or iterate over bidder configs checking if they include definition for slots icp or featured? This way we will have a single source of truth.

File: extensions/wikia/AdEngine3/src/setup.js [hunk 2]
```diff
@@ -179,6 +179,13 @@ function setupAdContext(wikiContext, isOptedIn = false, geoRequiresConsent = tru
 		if (!isGeoEnabled('wgAdDriverPubMaticOutstreamCountries')) {
 			context.remove('bidders.prebid.pubmatic.slots.INCONTENT_PLAYER');
 		}
+
+		const unbiddedSlotName = hasFeaturedVideo ? 'INCONTENT_PLAYER' : 'FEATURED';
```
Comment (jacekbj) [inline extensions/wikia/AdEngine3/src/setup.js]: Can you please change the name to `notBiddedSlotName`? I think there is no verb `to unbid` :)

Comment (dmnsobczak) [reply extensions/wikia/AdEngine3/src/setup.js]: Done 👍

File: extensions/wikia/AdEngine3/src/setup.js (hunk 1 above)
Comment (dmnsobczak) [reply extensions/wikia/AdEngine3/src/setup.js]: Great idea, changed.

File: extensions/wikia/AdEngine3/src/setup.js [hunk 3]
```diff
@@ -179,6 +179,19 @@ function setupAdContext(wikiContext, isOptedIn = false, geoRequiresConsent = tru
...
+		const notBiddedSlotName = hasFeaturedVideo ? 'INCONTENT_PLAYER' : 'FEATURED';
+		let videoBidders = [];
+
+		Object.keys(context.get('bidders.prebid')).forEach((bidder) => {
+			if (context.get(`bidders.prebid.${bidder}.videoBidder`)) {
+				videoBidders.push(bidder);
+			}
+		});
+
+		videoBidders.forEach((bidder) => {
```
Comment (jacekbj) [inline extensions/wikia/AdEngine3/src/setup.js:192]: I wish we had unit tests for this 🙈

**Review by jacekbj — APPROVED** [review]

**Comment by fraszczakszymon** [comment]
What about: https://github.com/Wikia/ad-engine/blob/dev/src/ad-bidders/prebid/prebid-helper.ts#L32

**Pull request closed without merging**